packages/labs/task/src/task.ts

```diff
@@ -194,16 +226,11 @@ export class Task<T extends [...unknown[]] = any, R = any> {
     }
   }
 
-  private _isDirty(deps: T) {
-    let i = 0;
-    const previousDeps = this._previousDeps;
-    this._previousDeps = deps;
-    for (const dep of deps) {
-      if (notEqual(dep, previousDeps[i])) {
-        return true;
-      }
-      i++;
-    }
-    return false;
+  private _argsDirty(args?: T) {
+    const prev = this._previousArgs;
+    this._previousArgs = args;
+    return Array.isArray(args) && Array.isArray(prev)
```

Comment:
I think if `_previousArgs` is undefined (or maybe state is initial), this should always return true.

Reply:
Added protected `shouldRun` which checks initial state.

Reply:
ah, right